simphony/io/tests/test_h5_cuds.py

```diff
@@ -255,5 +259,85 @@ def tearDown(self):
             engine.close()
 
 
+class TestMaterialsOperations(unittest.TestCase):
+
+    def setUp(self):
+        self.temp_dir = tempfile.mkdtemp()
+        self.filename = os.path.join(self.temp_dir, 'test.cuds')
+        self.addTypeEqualityFunc(
+            Material, partial(compare_materials, testcase=self))
+
+        self.expected = Materials()
+        for i in xrange(5):
+            self.expected.add_material(
+                Material(description="Material {}".format(i),
+                         data=create_data_container()))
+
+    def tearDown(self):
+        shutil.rmtree(self.temp_dir)
+
+    def file_factory(self):
+        return H5CUDS.open(self.filename)
+
+    def cleanup(self):
+        os.remove(self.filename)
+
+    def test_add_get_materials(self):
+        handle = self.file_factory()
+        expected_materials_dict = {}
+        for material in self.expected.iter_materials():
+            expected_materials_dict[material.uid] = material
+
+        handle.add_materials(self.expected)
+
+        # Closing and reopening the file
+        handle.close()
+        handle = self.file_factory()
+
+        stored = handle.get_materials()
+
+        stored_materials = [material for material in stored.iter_materials()]
+
+        self.assertEqual(len(expected_materials_dict), len(stored_materials))
+        for material in stored_materials:
+            self.assertEqual(
+                material, expected_materials_dict[material.uid])
+
+        handle.close()
+        self.cleanup()
```

Comment:
The idea was to recreate the file for each test, so that the Materials (group) would be empty at start of each of these tests. ```self.cleanup()``` just deletes this file, so that a new empty one is created in the next test.